prjxray/db.py

```diff
@@ -85,8 +87,33 @@ def __init__(self, db_root):
 
                 self.site_types[site_type_name] = os.path.join(self.db_root, f)
 
+            if f.endswith('_required_features.fasm'):
+                part = f[:-len('_required_features.fasm')]
+
+                with open(os.path.join(self.db_root, f), "r") as fp:
+                    features = []
+                    for line in fp:
+                        line = line.strip()
+                        if len(line) > 0:
+                            features.append(line)
+
+                    per_part_features[part] = set(features)
+
         self.tile_types_obj = {}
 
+        # Find common required features for all supported parts
```

Comment:
Why separate into common and part specific at all at this point?

Reply:
Fixed this. For now if the part is not explicitly given, no required features are returned.